src/bionics.cpp

```diff
@@ -1491,6 +1496,10 @@ bool player::install_bionics(const itype &type, int skill_level)
             add_bionic("bio_earplugs"); // automatically add the earplugs, they're part of the same bionic
         } else if (bioid == "bio_sunglasses") {
 			add_bionic("bio_blindfold"); // automatically add the Optical Dampers, they're part of the same bionic
+        } else if (bioid == "bio_eye_optic" && g->u.has_trait( "HYPEROPIC" ) ) {
```

Comment:
Correct me if I'm wrong, but this only gets run once upon installation, correct? The way this is written it looks like it will work fine if you have either hyperopic or myopic alone, but if you have both it will remove hyperopic only.

Issue #14751 has someone with this exact situation.